contracts/ProcedureTable.sol

```diff
@@ -388,28 +435,21 @@ library ProcedureTable {
         return r;
     }
 
-    function insert(Self storage /* self */, bytes24 key, address value, uint256[] caps) internal returns (bool replaced) {
+    function insert(Self storage /* self */, bytes24 key, address value) internal returns (bool replaced) {
         // First we get retrieve the procedure that is specified by this key, if
         // it exists, otherwise the struct we create in memory is just
         // zero-filled.
         Procedure memory p = _getProcedureByKey(uint192(key));
         // We then write or overwrite the various properties
         p.location = value;
-        // we just copy in the table in verbatim as long as its length is less
-        // than 128 (arbitrary, but less than 256 minus room for other parameters)
-        if (caps.length > 128) {
-            revert();
-        }
-
-        // The capabilities are parsed here. We neeed to pass in the Procedure
-        // struct as solidity can't return complex data structures.
-        _parseCaps(p,caps);
+        // TODO: change this
+        p.caps = new Capability[](0);
 
         // If the keyIndex is not zero then that indicates that the procedure
         // already exists. In this case *WE HAVE NOT OVERWRITTEN * the values,
         // as we have not called _storeProcdure.
         if (p.keyIndex > 0) {
-            return true;
+            revert();
```

Comment:
Is there a reason for adding a revert here? Isn't storage not yet changed at this point?

Reply:
I think here I was trying to make it consistent with a previous error condition that reverted if there were too many caps. It's not critical in any way.

I think this relates to your previous comment too, reverts vs returns. Might need to be part of an issue on consistent error handling.

Reply:
Ok, makes sense. I'll make an issue for error handling then.